modules/ppcp-api-client/src/Endpoint/OrderEndpoint.php

```diff
@@ -194,7 +194,7 @@ public function create(
 	): Order {
 		$bearer = $this->bearer->bearer();
 		$data   = array(
-			'intent'              => ( $this->subscription_helper->cart_contains_subscription() || $this->subscription_helper->current_product_is_subscription() ) ? 'AUTHORIZE' : $this->intent,
+			'intent'              => apply_filters( 'woocommerce_paypal_payments_saved_payment_subscription_intent', $this->intent ),
```

Comment:
The filter name may be confusing. This function is not just about subscriptions, but someone may think that it can be used to modify intent of subscriptions without touching other orders.

Something like `woocommerce_paypal_payments_order_intent` would be better.